pybigquery/sqlalchemy_bigquery.py

```diff
@@ -260,24 +276,33 @@ def group_by_clause(self, select, **kw):
     # no way to tell sqlalchemy that, so it works harder than
     # necessary and makes us do the same.
 
-    _in_expanding_bind = re.compile(r" IN \((\[EXPANDING_\w+\](:[A-Z0-9]+)?)\)$")
-
-    def _unnestify_in_expanding_bind(self, in_text):
-        return self._in_expanding_bind.sub(r" IN UNNEST([ \1 ])", in_text)
+    __in_expanding_bind = _helpers.substitute_re_method(
+        fr" IN \((\["
+        fr"{'EXPANDING' if sqlalchemy.__version__ < '1.4' else 'POSTCOMPILE'}"
```

Comment:
This check will not work as expected with SQLAlchemy v 1.10+, it will recognize it as _older_ than `1.4`.

```py
>>> "1.10" < "1.4"
True
```
I mean, it will take years to reach that, and version `1.3` will not be supported anymore, but a concern here is that this pattern could be reused (possibly by other maintainers then) and gradually transform into `sqlalchemy.__version__ < '1.9'` at some other place. And _then_ we are one minor version away from a false failure.

It's a long shot, indeed, but I'd say we fix this now and be better than some of the Y2K folks. :laughing:

Reply:
Done